File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    public int tribonacci(int n) {
        //T(n) = T(n-3) + T(n-2) + T(n-1)
        // d = a+b+c;
        if(n<2){
            return n;
        }
        int a=0, b=1, c=1, d;
        while(n-->2){
            d = a+b+c;
            a=b;
            b=c;
            c=d; 
        }
        return c;
    }
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        T1 = 0
        T2 = 1
        T3 = 1
        curr = 0
        if n<3:
            if n==0:
                return 0
            else:
                return 1
        for i in range(3, n+1):
            curr = T1+T2+T3
            T1=T2
            T2=T3
            T3=curr
            # print(curr)
        return curr